osquery/tables/system/darwin/running_apps.mm

```diff
@@ -0,0 +1,70 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#import <AppKit/AppKit.h>
+#import <Foundation/Foundation.h>
+
+#include <osquery/tables.h>
+
+namespace osquery {
+namespace tables {
+
+QueryData genRunningApps(QueryContext& context) {
+  QueryData results;
+  if (!context.isAnyColumnUsed({"pid", "bundle_identifier", "is_active"})) {
+    return results;
+  }
+
+  NSWorkspace* workspace = [NSWorkspace sharedWorkspace];
+  NSRunningApplication* appInFocus = [workspace frontmostApplication];
+  int appInFocusPid = appInFocus.processIdentifier;
+  // If query contains "where is_active = 1", return app in focus without
+  // iterating through all apps
+  if (context.constraints.count("is_active") > 0 &&
+      context.constraints.at("is_active").exists(EQUALS)) {
+    Row r;
+    context.setIntegerColumnIfUsed(r, "pid", appInFocusPid);
+    if (appInFocus.bundleIdentifier != nil) {
+      context.setTextColumnIfUsed(
+          r,
+          "bundle_identifier",
+          std::string([appInFocus.bundleIdentifier UTF8String]));
+    } else {
+      context.setTextColumnIfUsed(r, "bundle_identifier", "");
+    }
+    context.setIntegerColumnIfUsed(r, "is_active", 1);
+    results.push_back(r);
+    return results;
+  }
+
+  NSArray* runningApplications = [workspace runningApplications];
+  for (NSRunningApplication* app in runningApplications) {
+    Row r;
+    context.setIntegerColumnIfUsed(r, "pid", app.processIdentifier);
+    if (app.bundleIdentifier != nil) {
+      context.setTextColumnIfUsed(
+          r,
+          "bundle_identifier",
+          std::string([app.bundleIdentifier UTF8String]));
+    } else {
+      context.setTextColumnIfUsed(r, "bundle_identifier", "");
+    }
+    if (app.processIdentifier != appInFocusPid) {
```

Comment:
If you use `!app.isActive` here, you can factor the body of this loop into a helper function, and then use that within the special case as well.

Or even better:

`context.setIntegerColumnIfUsed(r, "is_active", app.isActive ? 1 : 0);`